src/components/burger-ingredients/options.tsx

```diff
@@ -5,72 +5,58 @@ import Separator from './separator';
 import IngredientDetails from '../ingredient-details/ingredient-details';
 import type { IngredientShape } from '../../utils/types.js'
 
-interface BurgerIngredientsProps{
+interface BurgerIngredientsProps {
   data: IngredientShape[],
   cart: IngredientShape[],
-  setCart: (newCart: IngredientShape[]|[]) => void,
+  setCart: (newCart: IngredientShape[]) => void,
+}
+
+type Sections = {
+  'bun': React.ReactNode[] | null,
+  'sauce': React.ReactNode[] | null,
+  'main': React.ReactNode[] | null,
 }
 
 const Options = (props: BurgerIngredientsProps) => {
-  const [detailsIngredient, setDetailsIngredient] = React.useState({} as IngredientShape);
+  const [detailsIngredient, setDetailsIngredient] = React.useState<IngredientShape | null>(null);
   const [showDetails, setShowDetails] = React.useState(false);
 
-  const orderFunc = (a:IngredientShape, b:IngredientShape) : number => {
-    const ORDER = ['bun', 'sauce', 'main'];
-    const index_a = ORDER.indexOf(a.type);
-    const index_b = ORDER.indexOf(b.type);
-    if (index_a < index_b) { return -1; }
-    if (index_a > index_b) { return 1; }
-    return 0;
-  }
-
-  const addToCart = (newIngredient:IngredientShape) => {
-    let newCart = props.cart.slice();
-    newCart.push(newIngredient);
-    props.setCart(newCart);
-  }
-
-  const SEP = [
-    {id: 'bun', name: 'Булки'},
-    {id: 'sauce', name: 'Соусы'},
-    {id: 'main', name: 'Начинки'},
-  ]
-
-  const template = Object.fromEntries( Object.entries(props.data[0]).map( ([k, v]) => [k, (typeof v == 'string'?"":0)]) ) as unknown as IngredientShape;
-  let options = [ ...props.data ];
-  options.sort(orderFunc);
-
-  for (const sep of SEP) {
-    const idx = options.findIndex( el => el.type === sep.id );
-    options.splice(idx, 0, {...template, _id:sep.id, type:"separator", name:sep.name}); //
-  }
-
-  const handelIngredientClick = (id:string) => {
-    setDetailsIngredient(props.data.filter(el => el._id === id)[0]);
+  const handelIngredientClick = (id: string) => {
+    let newIngredient = props.data.find(el => el._id === id);
+    setDetailsIngredient(newIngredient ? newIngredient : null);
```

Comment:
Можно лучше:
если переменная не меняет совего значения, то ее можно объявить с помощью `const`
Также чтобы не проверять на ее наличие можно воспользоваться таким приемом при объявлении `newIngredient`, если в случае ненахождения ингредиента значение было `null`: 
```suggestion
    const newIngredient = props.data.find(el => el._id === id) ?? null;
    setDetailsIngredient(newIngredient);
```